wkcuber/api/Layer.py

```diff
@@ -199,30 +199,36 @@ def _pad_existing_mags_for_downsampling(
 
     def downsample(
         self,
-        from_mag: Mag,
-        max_mag: Optional[Mag],
-        interpolation_mode: str,
-        compress: bool,
+        from_mag: Optional[Mag] = None,
+        max_mag: Optional[Mag] = None,
```

Comment:
I'm still a bit confused about the semantics of the following parameters:

- max_mag
- anisotropic
- scale

I think the major use case for `max_mag` is to tell the cuber to stop after magnification X. However, it can also be used to determine the exact target mag (i.e., it prescribes the exact downsampling strategy) *if* `anisotropic` is set to True, correct? In that case a provided `scale` is overridden (I think, there should be a warning or maybe even a raised assertion for that).
I'm not sure the latter behavior of `max_mag` is intuitive. Maybe this should be a separate parameter? (e.g., `target_mag`).

What happens if I explicitly set `anisotropic` to `False` ? If I'm not mistaken, this will still use the DS scale and create anisotropic magnifications which is probably not what one would expect when passing `False`.

Regarding the scale: What is the usecase for providing this if the dataset's metadata scale exists? If one wants to mess with the downsampling steps, it's probably easier to pass a target mag.

My suggestion would be:
- let's make sure the parameters all have a helpful and easy-to-understand description which also mentions interdependencies between the parameters
- if it's possible to pass incompatible parameters (e.g., pass a scale, but also pass a target mag?), throw an error
- make sure the interface as a whole makes sense. maybe we can simplify some things. you probably have a better picture of this in your head than me :)

Reply:
You are right. Here is an updated version:
```
      if anisotropic is not None:
          if anisotropic:
              assert scale is not None  # The user has to specify an anisotropic scale
          else:
              scale = (2, 2, 2)
      else:
          if scale is None:
              scale = self.dataset.properties.scale
``` 

However, this raises the question if `anisotropic` as a parameter is really necessary. What if we remove `anisotripic` completely?  If the dataset has an anisotropic scale, but the user want to downsample isotropic, then he has to specify `scale=(2,2,2)`
```
if scale is None:
    scale = self.dataset.properties.scale
# otherwise the parameter is used
```